apps/web-giddh/src/app/gst/gst-setting/gst-setting.component.ts

```diff
@@ -0,0 +1,339 @@
+            let lastFormArray = mappings.at(lastIndex);
+            let fromDate = null;
+            let toDate = null;
+            if (user.fromDate && user.toDate) {
+                fromDate = dayjs(user?.fromDate, GIDDH_DATE_FORMAT).format(GIDDH_DATE_FORMAT_YYYY_MM_DD);
+                toDate = dayjs(user.toDate, GIDDH_DATE_FORMAT).format(GIDDH_DATE_FORMAT_YYYY_MM_DD);
+            }
+            if (!lastFormArray?.get('lutNumber')?.value) {
+                mappings.at(lastIndex).patchValue({
+                    fromDate: fromDate,
+                    toDate: toDate,
+                    lutNumber: user.lutNumber,
+                    uniqueName: user.uniqueName
+                });
+            }
+        } else {
+            mappings.push(this.initLutForm({
+                fromDate: this.fromDate,
+                toDate: this.toDate
+            }));
+        }
+    }
+
+    /**
+     * This will be use for remove lut item
+     *
+     * @param {number} index
+     * @memberof GstSettingComponent
+     */
+    public removeLutItem(index: number): void {
+        let mappings = this.gstSettingForm.get('gstData') as FormArray;
+        let mappingForm = mappings.at(index);
+        if (index === 0) {
+            if (mappingForm.get('uniqueName')?.value) {
+                this.componentStore.deleteLutNumber({ lutNumberUniqueName: mappingForm.get('uniqueName')?.value });
+                mappings.removeAt(index);
+                mappings.push(this.initLutForm({
+                    fromDate: this.fromDate,
+                    toDate: this.toDate
+                }));
+                this.responseArray[index]['message'] = null;
+            } else {
+                mappingForm.get('lutNumber')?.patchValue(null);
+                mappingForm.get('fromDate')?.patchValue(this.fromDate);
+                mappingForm.get('toDate')?.patchValue(this.toDate);
+                this.responseArray[index]['message'] = null;
+            }
+        } else {
+            mappings.removeAt(index);
+            if (mappingForm.get('uniqueName')?.value) {
+                this.responseArray[index] = [];
+                this.componentStore.deleteLutNumber({ lutNumberUniqueName: mappingForm.get('uniqueName')?.value });
+            }
+        }
+    }
+
+    /**
+    * Callback for translation response complete
+    *
+    * @param {*} event
+    * @memberof GstSettingComponent
+    */
+    public translationComplete(event: any): void {
+        if (event) {
+            this.exportTypes = [
+                { label: this.localeData?.with_pay, value: 'yes' },
+                { label: this.localeData?.without_pay, value: 'no' }
+            ];
+            this.changeDetection.detectChanges();
+        }
+    }
+
+    /**
+     *This will be use for create update lut form
+     *
+     * @memberof GstSettingComponent
+     */
+    public saveLutNumbers(): void {
+        if (!this.gstSettingForm.dirty) {
+            return;
```

Comment:
show some message here otherwise user will think nothing is happening. Confirm with tester

Reply:
Confirmed with Kriti, we are not showing any message